.github/workflows/main.yml

```diff
@@ -0,0 +1,55 @@
+name: Build DAPHNE
+
+on:
+  workflow_dispatch:
+  push:
+
+jobs:
+  build:
+    runs-on: ubuntu-22.04
+    env:
+      # fix usage of tput in build script:
+      TERM: dumb
+      DEBIAN_FRONTEND: noninteractive
+
+    steps:
+    -
+      name: "Install build dependencies"
+      run: |
+        set -e
+        pwd
+        echo Installing ...
+        sudo apt-get update
+        sudo apt-get install -y cmake clang default-jdk git libssl-dev lld pkg-config python3 python3-pip gfortran uuid-dev libboost-dev unzip wget moreutils
+        sudo wget -qO /usr/local/bin/ninja.gz https://github.com/ninja-build/ninja/releases/latest/download/ninja-linux.zip
+        sudo gunzip /usr/local/bin/ninja.gz
+        sudo chmod a+x /usr/local/bin/ninja
+        sudo python3 -m pip install -U numpy
+        echo "---------"
+        ninja --version
+        python3 -m pip show numpy
+
+    -
+      name: Checkout
+      uses: actions/checkout@v3
+    -
+      name: Building
+      run: |
+        set -e
+        ./build.sh -nf --clean --yes | ts
+        ./build.sh -nf | ts
+    -
+      name: Testing
+      run: ./test.sh | ts
+    -
+      name: List
+      run: |
+        ls -la bin
+    -
+      name: 'Upload Artifact'
+      uses: actions/upload-artifact@v3
+      with:
+        name: daphne
+        path: |
+          bin/daphne
+          bin/DistributedWorker
```

Comment:
What is this list and upload for (sorry - not an expert on Github Workflows/Actions) ?

Reply:
list was a debugging step (are binaries generated)
upload is for storing the generated binaries (without the test_runner) for downloading - using Github to build the binary instead of locally ... (accessible for contributors only, for public, a release must be created)

Reply:
OK this sounds nice :) Please include the lib directory as well to have a fully working DAPHNE build. Btw does this work with wild cards (e.g., bin/* lib/*) ?

Reply:
ok I see. I assumed you statically link.
the star would not be possible.
I can provide an update tomorrow, have to leave for an appointment now ;)